youtube_dl/extractor/generic.py

```diff
@@ -2009,15 +2009,17 @@ def _extract_rss(self, url, video_id, doc):
 
         entries = []
         for it in doc.findall('./channel/item'):
-            next_url = xpath_text(it, 'link', fatal=False)
-            if not next_url:
-                enclosure_nodes = it.findall('./enclosure')
-                for e in enclosure_nodes:
-                    next_url = e.attrib.get('url')
-                    if next_url:
-                        break
-
-            if not next_url:
+            next_url = None
+            enclosure_nodes = it.findall('./enclosure')
+            for e in enclosure_nodes:
+                next_url = e.attrib.get('url')
+                if next_url:
+                    break
+
+            if not enclosure_nodes:
+                next_url = xpath_text(it, 'link', fatal=False)
+
+            if next_url is None:
```

Comment:
This won't skip empty strings.

Reply:
i changed it back to `not next_url`. This should do the job in conjunction with the `next_url = None` from above.